File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        map: Dict = {}
        for num in nums:
            map[num] = map.get(num, 0) + 1
        for (key,value) in map.items():
            if value >= 2:
                return True
        return False
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.swift
class Solution {
    func topKFrequent(_ nums: [Int], _ k: Int) -> [Int] {
        var map: [Int:Int] = [:]
        var ans: [Int] = []
        for num in nums {
            map[num,default: 0] += 1
        }
        var arr = map.map{ ($0.key,$0.value)}
        sort(&arr)

        for i in 0..<k {
            ans.append(arr[i].0)
        }
        return ans

    }

    func sort( _ arr: inout [(Int,Int)]){
        for i in 0..<arr.count {
            for j in (i + 1)..<arr.count {
                if arr[i].1 < arr[j].1 {
                    arr.swapAt(i,j)
                }
            }
        }
    }
}
